concent_api/core/views.py

```diff
@@ -1220,6 +1225,632 @@ def store_message_and_message_status(
+
+    subtask = Subtask(
+        task_id         = task_id,
+        subtask_id      = subtask_id,
+        provider        = provider,
+        requestor       = requestor,
+        state           = state.name,
+        next_deadline   = parse_timestamp_to_utc_datetime(next_deadline),
+    )
+
+    set_subtask_messages(
+        subtask,
+        task_to_compute             = task_to_compute,
+        report_computed_task        = report_computed_task,
+        ack_report_computed_task    = ack_report_computed_task,
+        reject_report_computed_task = reject_report_computed_task,
+        subtask_results_accepted    = subtask_results_accepted,
+        subtask_results_rejected    = subtask_results_rejected,
+    )
+
+    subtask.full_clean()
+    subtask.save()
+
+    logging.log_subtask_stored(
+        task_id,
+        subtask_id,
+        state.name,
+        subtask.provider.public_key,
+        subtask.requestor.public_key,
+        next_deadline,
+    )
+
+    return subtask
+
+
+def verify_file_status(
+    client_public_key: bytes,
+):
+    """
+    Function to verify existence of a file on cluster storage
+    """
+
+    force_get_task_result_list = PendingResponse.objects.filter(
+        subtask__requestor__public_key  = b64encode(client_public_key),
+        subtask__state                  = Subtask.SubtaskState.FORCING_RESULT_TRANSFER.name,  # pylint: disable=no-member
+    )
+
+    for pending_reponse in force_get_task_result_list:
+        report_computed_task    = deserialize_message(pending_reponse.subtask.report_computed_task.data.tobytes())
+        file_transfer_token     = create_file_transfer_token(
+            report_computed_task,
+            client_public_key,
+        )
+        if get_file_status(file_transfer_token):
+            subtask               = pending_reponse.subtask
+            subtask.state         = Subtask.SubtaskState.RESULT_UPLOADED.name  # pylint: disable=no-member
+            subtask.next_deadline = None
+            subtask.full_clean()
+            subtask.save()
+
+            store_pending_message(
+                response_type       = PendingResponse.ResponseType.ForceGetTaskResultDownload, # TODO DOWNLOAD
+                client_public_key   = subtask.requestor.public_key_bytes,
+                queue               = PendingResponse.Queue.Receive,
+                subtask             = subtask,
+            )
+            logging.log_file_status(
+                subtask.subtask_id,
+                subtask.requestor.public_key,
+                subtask.provider.public_key,
+            )
+
+
+def update_timed_out_subtasks(
+    client_public_key: bytes,
+):
+    verify_file_status(client_public_key)
+
+    clients_subtask_list = Subtask.objects.filter(
+        Q(requestor__public_key = b64encode(client_public_key)) | Q(provider__public_key = b64encode(client_public_key)),
+        state__in               = [state.name for state in Subtask.ACTIVE_STATES],
```

Comment:
You could do
``` python
state__in = list(Subtask.ACTIVE_STATES),
```

or maybe even (I don't see why `__in` would not work with `set` type but I have not checked so I may be wrong here):
``` python
state__in = Subtask.ACTIVE_STATES,
```

Reply:
None of these two way works.

Reply:
Oh, sorry, my bad. I did not notice that you need names there. What I proposed does not work because it's a list of enum items, not names.